source/core/parameters.cc

```diff
@@ -2200,8 +2199,65 @@ namespace Parameters
           std::string shape_arguments_str = prm.get("shape arguments");
           std::vector<std::string> shape_arguments_str_list(
             Utilities::split_string_list(shape_arguments_str, ";"));
-          std::vector<double> shape_arguments =
-            Utilities::string_to_double(shape_arguments_str_list);
+          std::vector<double> shape_arguments;
+          if (shape_type == "rbf")
+            {
+              if constexpr (dim == 3)
+                {
+                  std::string shape_name = shape_arguments_str_list[0];
+                  std::map<std::string, std::vector<double>> rbf_data;
+                  fill_vectors_from_file(rbf_data, shape_name, ",");
+                  size_t number_of_nodes = rbf_data["weight"].size();
+
+                  std::vector<double> weights;
+                  std::vector<double> basis_functions;
+                  std::vector<double> support_radii;
+                  std::vector<double> nodes_x;
+                  std::vector<double> nodes_y;
+                  std::vector<double> nodes_z;
+
+                  weights.resize(number_of_nodes);
+                  basis_functions.resize(number_of_nodes);
+                  support_radii.resize(number_of_nodes);
+                  nodes_x.resize(number_of_nodes);
+                  nodes_y.resize(number_of_nodes);
+                  nodes_z.resize(number_of_nodes);
+
+                  for (size_t n_i = 0; n_i < number_of_nodes; n_i++)
+                    {
+                      weights[n_i]         = rbf_data["weight"][n_i];
+                      support_radii[n_i]   = rbf_data["support_radius"][n_i];
+                      basis_functions[n_i] = rbf_data["basis_function"][n_i];
+                      nodes_x[n_i]         = rbf_data["xnode"][n_i];
+                      nodes_y[n_i]         = rbf_data["ynode"][n_i];
+                      nodes_z[n_i]         = rbf_data["znode"][n_i];
+                    }
```

Comment:
Is this intermediary generation of a vector necessary? I Mean it's not a big deal, but at least please comment what you are doing.